### PR TITLE
M5: add Docker to project and update readme

### PR DESCRIPTION
**Pull request by Lucas-Fernandes-Nogueira**
I've done the 3 parts and I think everything is working! Thanks for your help, Hugh 👍

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,41 @@ If you have Vagrant installed in your machine, provision and run a virtual machi
...
+### Dev
+In developer mode, Flask tracks changes to the files in the codebase. You should see the changes after a simple tab refresh, without the need to rebuild the docker image.
+
+To run in developer mode, first build the docker image with the following command from the project directory:
+```bash
+docker build --target development --tag todo-app:dev .
+```
+After you've built the image, run the website with the following command:
+```bash
+docker run -p 5000:5000 --env-file .env --mount type=bind,source="$(pwd)",destination=/todo-app todo-app:dev
```
Comment (hugh-emerson) [inline README.md]: You don't actually need `--env-file .env` if you are mounting the whole directory

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.8.6-buster as base
+RUN pip install "poetry==1.1.4"
+WORKDIR /todo-app
+EXPOSE 5000
+
+FROM base as production
+COPY . /todo-app
```
Comment (hugh-emerson) [inline Dockerfile]: To improve build layer caching, you can split this into copying the poetry files here, and then copying `.` after the poetry install.

File: .gitignore [hunk 3]
```diff
@@ -1,126 +1,15 @@
...
-.eggs/
-lib/
-lib64/
-parts/
-sdist/
-var/
-wheels/
-pip-wheel-metadata/
-share/python-wheels/
-*.egg-info/
```
Comment (hugh-emerson) [inline .gitignore:34]: I believe doing `poetry install` will create an .egg-info folder, unless you run it with --no-root

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+#VSCode folder
+.vscode
+
+#configuration files
+.env
+.env.template
+
+#gitignore
+.gitignore
```
Comment (hugh-emerson) [inline .dockerignore:13]: Pretty reasonable list, but how about `.git`?

**Review by hugh-emerson — APPROVED** [review]
Looks good 👍 I've just left some minor comments, merge whenever you're ready

File: .dockerignore (hunk 4 above)
Comment (Lucas-Fernandes-Nogueira) [reply .dockerignore:13]: Done :)

File: .gitignore (hunk 3 above)
Comment (Lucas-Fernandes-Nogueira) [reply .gitignore:34]: Thanks!

File: Dockerfile (hunk 2 above)
Comment (Lucas-Fernandes-Nogueira) [reply Dockerfile]: Done!

File: README.md (hunk 1 above)
Comment (Lucas-Fernandes-Nogueira) [reply README.md]: Thanks, I didn't spot that!

**Pull request merged**